src/pip/_internal/operations/prepare.py

```diff
@@ -498,27 +523,30 @@ def _prepare_linked_requirement(self, req, parallel_builds):
 
         self._ensure_link_req_src_dir(req, parallel_builds)
         hashes = self._get_linked_req_hashes(req)
-        if link.url not in self._downloaded:
-            try:
-                local_file = unpack_url(
-                    link, req.source_dir, self._download,
-                    self.download_dir, hashes,
-                )
-            except NetworkConnectionError as exc:
-                raise InstallationError(
-                    'Could not install requirement {} because of HTTP '
-                    'error {} for URL {}'.format(req, exc, link)
-                )
-        else:
-            file_path, content_type = self._downloaded[link.url]
-            if hashes:
-                hashes.check_against_path(file_path)
-            local_file = File(file_path, content_type)
-
-        # For use in later processing,
-        # preserve the file path on the requirement.
-        if local_file:
-            req.local_file_path = local_file.path
+
+        if not (link.is_existing_dir() and self.in_tree_build):
```

Comment:
nit: I'd find this slightly more readable, but it might be a matter of preference

```
if link.is_existing_dir() and self.in_tree_build:
    local_file = None
elif link.url not in self._downloaded:
    ...
```

Reply:
👍 agreed